airflow/timetables/interval.py

```diff
@@ -86,6 +79,13 @@ class CronDataIntervalTimetable(_DataIntervalTimetable):
     def __init__(self, cron: str, timezone: datetime.tzinfo) -> None:
         self._schedule = CronSchedule(cron, timezone)
 
+    def infer_data_interval(self, run_after: DateTime) -> DataInterval:
+        # Get the last complete period before run_after, e.g. if a DAG run is
+        # scheduled at each midnight, the data interval of a manually triggered
+        # run at 1am 25th is between 0am 24th and 0am 25th.
+        end = self._schedule.get_prev(self._schedule.align(run_after))
+        return DataInterval(start=self._schedule.get_prev(end), end=end)
+
 
 class DeltaDataIntervalTimetable(_DataIntervalTimetable):
```

Comment:
(not related to this PR)

Should we change the name to `DataIntervalWithDeltaTimetable` ? no strong opinion

Reply:
No strong opinion from me either, this name is not intended to be used directly by users anyway (they can just use `schedule_interval`). The only “complaint” (a weird term considering I named these classes) I have to this (and `CronDataIntervalTimetable`) is they are so terribly long.